### PR TITLE
Separate CartSharedModule with other libs

### LINKED ISSUE #13072: Validate that we don't need to duplicate the shared cart component ( cart item list )
Validate that all works well with other libs depending on the cart lib

Order lib
Order approvals

### PR DESCRIPTION
**Pull request by WeizhengSap**
close #13072

### CODE REVIEW

File: feature-libs/cart/main/components/cart/cart-shared/order-summary/order-summary.component.html [hunk 1]
```diff
@@ -58,13 +58,6 @@ <h3>{{ 'orderCost.orderSummary' | cxTranslate }}</h3>
   </ng-template>
 </div>
 
-<!--
-<cx-promotions
-  *ngIf="cart"
-  [promotions]="cart.appliedOrderPromotions"
-></cx-promotions>
--->
-
```
Comment (plabadie) [inline feature-libs/cart/main/components/cart/cart-shared/order-summary/order-summary.component.html:67]: Don't we need something to replace cx-promotions?

File: feature-libs/cart/main/components/cart/cart-shared/cart-item-list/cart-item-list.component.ts [hunk 2]
```diff
@@ -58,24 +61,55 @@ export class CartItemListComponent implements OnInit, OnDestroy {
...
   }
 
   constructor(
     protected activeCartService: ActiveCartFacade,
     protected selectiveCartService: SelectiveCartFacade,
     protected userIdService: UserIdService,
-    protected multiCartService: MultiCartFacade
+    protected multiCartService: MultiCartFacade,
+    protected cd: ChangeDetectorRef,
+    @Optional() protected outlet?: OutletContextData
```
Comment (plabadie) [inline feature-libs/cart/main/components/cart/cart-shared/cart-item-list/cart-item-list.component.ts]: Is it possible to specify the generics type of OutletContextData ini this case? ( like OutletContextData &lt;SomeType&gt; )

File: feature-libs/cart/saved-cart/components/details/saved-cart-details.module.ts [hunk 3]
```diff
@@ -24,16 +21,15 @@ import { SavedCartDetailsOverviewComponent } from './saved-cart-details-overview
...
 @NgModule({
   imports: [
-    CartSharedModule,
     CommonModule,
     I18nModule,
     UrlModule,
     RouterModule,
     CardModule,
     MediaModule,
-    AddToCartModule,
```
Comment (plabadie) [inline feature-libs/cart/saved-cart/components/details/saved-cart-details.module.ts:34]: `AddToCartModule` was not used?

File: feature-libs/cart/main/components/cart/cart-shared/cart-shared.module.ts [hunk 4]
```diff
@@ -34,6 +35,16 @@ import { OrderSummaryComponent } from './order-summary/order-summary.component';
     ModalModule,
     OutletModule,
   ],
+  providers: [
+    provideOutlet({
+      id: 'cx-order-summary',
```
Comment (plabadie) [inline feature-libs/cart/main/components/cart/cart-shared/cart-shared.module.ts]: Would it be relevant to use constants or enums for the outlet names?  Other parrts of the codebase seem tu use enums.

File: feature-libs/cart/main/components/cart/cart-shared/order-summary/order-summary.component.html (hunk 1 above)
Comment (WeizhengSap) [reply feature-libs/cart/main/components/cart/cart-shared/order-summary/order-summary.component.html:67]: I just removed the commented codes, nothing else.

File: feature-libs/cart/saved-cart/components/details/saved-cart-details.module.ts (hunk 3 above)
Comment (WeizhengSap) [reply feature-libs/cart/saved-cart/components/details/saved-cart-details.module.ts:34]: Not used.

File: feature-libs/cart/main/components/cart/cart-shared/cart-item-list/cart-item-list.component.ts (hunk 2 above)
Comment (WeizhengSap) [reply feature-libs/cart/main/components/cart/cart-shared/cart-item-list/cart-item-list.component.ts]: I can give it `any`, or something like this `{readonly: boolean, ....}` (wrap all inputs into one type).

Comment (plabadie) [reply feature-libs/cart/main/components/cart/cart-shared/cart-item-list/cart-item-list.component.ts]: In general we try not to use `any` when we can use a more precise alternative, but your call as to wether you think it's relevant here.

File: feature-libs/cart/saved-cart/components/details/saved-cart-details.module.ts (hunk 3 above)
Comment (plabadie) [reply feature-libs/cart/saved-cart/components/details/saved-cart-details.module.ts:34]: Good catch

File: feature-libs/cart/main/components/cart/cart-shared/order-summary/order-summary.component.html (hunk 1 above)
Comment (plabadie) [reply feature-libs/cart/main/components/cart/cart-shared/order-summary/order-summary.component.html:67]: Ok, I thought we had commented this out in earlier commits of the cart-lib epic.

**Review by plabadie — APPROVED** [review]

**Pull request merged**